include/ada/url-inl.h

```diff
@@ -96,39 +100,39 @@ inline std::ostream &operator<<(std::ostream &out, const ada::url &u) {
     running_index += helpers::fast_digit_count(*port) + 1;  // Port omits ':'
   }
 
-  out.pathname_start = uint32_t(running_index);
+  out.pathname_start = static_cast<uint32_t>(running_index);
 
   running_index += path.size();
 
   if (query.has_value()) {
-    out.search_start = uint32_t(running_index);
+    out.search_start = static_cast<uint32_t>(running_index);
     running_index += get_search().size();
-    if (get_search().size() == 0) {
+    if (get_search().empty()) {
       running_index++;
     }
   }
 
   if (hash.has_value()) {
-    out.hash_start = uint32_t(running_index);
+    out.hash_start = static_cast<uint32_t>(running_index);
   }
 
   return out;
 }
 
 inline void url::update_base_hostname(std::string_view input) { host = input; }
 
-inline void url::update_unencoded_base_hash(std::string_view input) {
+inline void url::update_unencoded_base_hash(std::string_view const input) {
   // We do the percent encoding
   hash = unicode::percent_encode(input,
                                  ada::character_sets::FRAGMENT_PERCENT_ENCODE);
 }
 
-inline void url::update_base_search(std::string_view input,
+inline void url::update_base_search(std::string_view const input,
```

Comment:
Why would you ever pass a `const std::string_view` as opposed to a `std::string_view`?

Reply:
The same reason why we use `const` in variables. To help the compiler and also ourselves that the `input` is not gonna be used as a mutable variable during the body.